src/openrct2/interface/Viewport.cpp

```diff
@@ -543,10 +543,10 @@ void viewport_update_position(rct_window* window)
 
     viewport_set_underground_flag(0, window, viewport);
 
-    int16_t x = window->savedViewPos.x + viewport->view_width / 2;
-    int16_t y = window->savedViewPos.y + viewport->view_height / 2;
+    auto coords = ScreenCoordsXY{ window->savedViewPos.x + viewport->view_width / 2,
+                                  window->savedViewPos.y + viewport->view_height / 2 };
```

Comment:
As suggested by Duncan previously:
```suggestion
    auto viewportMidPoint = ScreenCoordsXY{ window->savedViewPos.x + viewport->view_width / 2,
                                            window->savedViewPos.y + viewport->view_height / 2 };
```